### PR TITLE
feat: fetch verification method types by proof type

### PR DESCRIPTION
**Pull request by karimStekelenburg**
Signed-off-by: Karim <karim@animo.id>

### CODE REVIEW

File: packages/core/src/modules/vc/W3cCredentialService.ts [hunk 1]
```diff
@@ -352,6 +352,10 @@ export class W3cCredentialService {
     return result.map((record) => record.credential)
   }
 
+  public getKeyTypeByProofType(proofType: string): KeyType {
+    return this.suiteRegistry.getKeyTypeByProofType(proofType)
+  }
```
Comment (TimoGlastra) [inline packages/core/src/modules/vc/W3cCredentialService.ts:357]: a proof type could potentially have multiple supported key types associated with it. Should we make this an array?

**Review by TimoGlastra — CHANGES_REQUESTED** [review]
Can you add tests for the added methods?

File: packages/core/src/modules/vc/W3cCredentialService.ts (hunk 1 above)
Comment (karimStekelenburg) [reply packages/core/src/modules/vc/W3cCredentialService.ts:357]: Good point, that will cause issues later on. Resolved now.

**Comment by karimStekelenburg** [comment]
> Can you add tests for the added methods?

@TimoGlastra good catch, thanks. Time for a PR checklist.. I'll open a discussion about it.

File: packages/core/src/modules/vc/SignatureSuiteRegistry.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export interface SuiteInfo {
   suiteClass: typeof LinkedDataSignature
   proofType: string
   requiredKeyType: string
-  keyType: string
+  keyTypes: [KeyType]
```
Comment (TimoGlastra) [inline packages/core/src/modules/vc/SignatureSuiteRegistry.ts:14]: This should be `KeyType[]`. This actually means an array with one item of type `KeyType` (you can use this to create tuple types)

```suggestion
  keyTypes: KeyType[]
```

Comment (TimoGlastra) [reply packages/core/src/modules/vc/SignatureSuiteRegistry.ts:14]: Hmm after thinking about it a bit more. This is the key type as we use it in the framework, but not the key as is used in did documents. Those key types are e.g. Ed25519Key2018 (or something like it). So we still have to make the transformation, but only the signature suites know which actual types is supports. 

Should this be updated to include the supported key types as used by did documents? E.g. if the suites only supports 2018 and 2020 ed25519 keys and there's a new 2022 one that isn't supported we should only extract the 2018 and 2022 keys from the did document, as that's the only ones we can sign with.

**Comment by TimoGlastra** [comment]
BBS tests failing in Node.JS 17/18 But otherwise tests are passing. Merging

**Pull request merged**